src/backend/crossterm.rs

```diff
@@ -4,28 +4,188 @@
+        );
+
+        match event {
+            // Handle Char + modifier.
+            CKeyEvent {
+                modifiers: KeyModifiers::CONTROL,
+                code: KeyCode::Char('c'),
+            } => Event::Exit,
+            CKeyEvent {
+                modifiers: KeyModifiers::CONTROL,
+                code: KeyCode::Char(c),
+            } => Event::CtrlChar(c),
+            CKeyEvent {
+                modifiers: KeyModifiers::ALT,
+                code: KeyCode::Char(c),
+            } => Event::AltChar(c),
+            CKeyEvent {
+                modifiers: KeyModifiers::SHIFT,
+                code: KeyCode::Char(c),
+            } => Event::Char(c),
+
+            // Handle key + multiple modifiers
+            CKeyEvent {
+                modifiers: CTRL_ALT,
+                code,
+            } => Event::CtrlAlt(Key::from(code)),
+            CKeyEvent {
+                modifiers: CTRL_SHIFT,
+                code,
+            } => Event::CtrlShift(Key::from(code)),
+            CKeyEvent {
+                modifiers: ALT_SHIFT,
+                code,
+            } => Event::AltShift(Key::from(code)),
+
+            // Handle key + single modifier
+            CKeyEvent {
+                modifiers: KeyModifiers::CONTROL,
+                code,
+            } => Event::Ctrl(Key::from(code)),
+            CKeyEvent {
+                modifiers: KeyModifiers::ALT,
+                code,
+            } => Event::Alt(Key::from(code)),
+            CKeyEvent {
+                modifiers: KeyModifiers::SHIFT,
+                code,
+            } => Event::Shift(Key::from(code)),
+
+            CKeyEvent {
+                code: KeyCode::Char(c),
+                ..
+            } => Event::Char(c),
+
+            // Explicitly handle 'backtab' since crossterm does not sent SHIFT alongside the back tab key.
+            CKeyEvent {
+                code: KeyCode::BackTab,
+                ..
+            } => Event::Shift(Key::Tab),
+
+            // All other keys.
+            CKeyEvent { code, .. } => Event::Key(Key::from(code)),
+        }
+    }
+}
+
+impl From<theme::Color> for Color {
+    fn from(base_color: theme::Color) -> Self {
+        match base_color {
+            theme::Color::Dark(theme::BaseColor::Black) => Color::Black,
+            theme::Color::Dark(theme::BaseColor::Red) => Color::DarkRed,
+            theme::Color::Dark(theme::BaseColor::Green) => Color::DarkGreen,
+            theme::Color::Dark(theme::BaseColor::Yellow) => Color::DarkYellow,
+            theme::Color::Dark(theme::BaseColor::Blue) => Color::DarkBlue,
+            theme::Color::Dark(theme::BaseColor::Magenta) => {
+                Color::DarkMagenta
+            }
+            theme::Color::Dark(theme::BaseColor::Cyan) => Color::DarkCyan,
+            theme::Color::Dark(theme::BaseColor::White) => Color::Grey,
+            theme::Color::Light(theme::BaseColor::Black) => Color::Grey,
```

Comment:
Note that "dark white" and "light dark" are two different colors, index 7 and 8 respectively in the palette.

Reply:
ye, this would be `Color::DarkGrey`

Reply:
Or maybe `Color::DarkWhite`?

Reply:
Dark white does not exist in crossterm, this would just be grey. This should be DarkGrey as @vmedea  said.

Reply:
Ah, I was looking at the documentation, but I guess it's outdated:

https://docs.rs/crossterm/0.14.1/crossterm/style/enum.Color.html#platform-specific-notes

Reply:
Thanks, I will have to update that.